File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.cpp
class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& values) {
        int iMax = values[0] + 0; int ans = 0;
        for(int j = 1; j < values.size(); ++j) {
            ans = max(ans, iMax + values[j] - j);
            iMax = max(iMax, values[j] + j);
        }
        return ans;
    }
};

File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        iMax, ans = values[0], 0
        for i in range(1, len(values)):
            ans = max(ans, iMax + values[i]-i)
            iMax = max(iMax, values[i]+i)
        return ans